mlir-sycl/CMakeLists.txt

```diff
@@ -70,6 +70,23 @@ set(MLIR_SYCL_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin)
 set(LLVM_LIT_ARGS "-sv" CACHE STRING "lit default options")
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/modules")
 
+# Generate Documentation
+add_custom_target(sycl-doc)
```

Comment:
Do we really need the custom CMake logic? 

It seems there are already targets to build the different parts of the documentation, e.g.:
`ninja SYCLOpsDocGen` generates `SYCLOps.md` in the `doc/Dialect` folder inside the build directory for me. 

Alternatively, my build also has a target for the file itself: `ninja docs/Dialects/SYCLOps.md`

If we want a `sycl-doc` target, we could simply have that target depends on those targets and remove the custom commands and function here.

Reply:
I feel more inclined to having some logic to handle this, but it should probably be in its own CMake module in `mlir-sycl/cmake/modules/AddMLIRSYCL.cmake`. We can alternatively have:
```cmake
function(add_mlir_sycl_doc doc_filename output_file output_directory command)
  add_mlir_doc(${doc_filename} ${output_file} ${output_directory} ${command})
  add_dependencies(mlir-sycl-doc ${output_file}DocGen)
endfunction()
```
Also, `sycl-doc` might not be the ideal name... can we have `mlir-sycl-doc` instead?

Reply:
`add_dependencies(mlir-doc mlir-sycl-doc)` wouldn't be needed this way

Reply:
> I feel more inclined to having some logic to handle this, but it should probably be in its own CMake module in `mlir-sycl/cmake/modules/AddMLIRSYCL.cmake`. We can alternatively have the following function to use the target created by `add_mlir_doc`:
> 
> ```cmake
> function(add_mlir_sycl_doc doc_filename output_file output_directory command)
>   add_mlir_doc(${doc_filename} ${output_file} ${output_directory} ${command})
>   add_dependencies(mlir-sycl-doc ${output_file}DocGen)
> endfunction()
> ```
> 
> Also, `sycl-doc` might not be the ideal name... can we have `mlir-sycl-doc` instead?

Having just this small wrapper seems fine, it would setup the necessary dependencies of the `sycl-doc` (or however we want to call it) target, without the need to copy large parts of `add_mlir_doc`.

Reply:
`sycl-doc` seems odd to me as there is already a `sycl` project. We might want to call this `mlir-sycl` instead

Reply:
Yes, I agree that it might be misleading, probably good to add `mlir` somewhere in the name.

Reply:
done.